mds.run_start_given_uid fails cryptically if given unknown uid

mds.run_start_given_uid(uid='some unknown uid') should fail better. Currently it fails here:

```python
    def _cache_run_start(run_start, run_start_cache):
        """De-reference and cache a RunStart document

        The de-referenced Document is cached against the
        ObjectId and the uid -> ObjectID mapping is stored.

        Parameters
        ----------
        run_start : dict
            raw pymongo dictionary. This is expected to have
            an entry `_id` with the ObjectId used by mongo.

        Returns
        -------
        run_start : doc.Document
            Document instance for this RunStart document.
            The ObjectId has been stripped.
        """
>       run_start = dict(run_start)
E       TypeError: 'NoneType' object is not iterable
```

--- a/metadatastore/commands.py
+++ b/metadatastore/commands.py
@@ -12,7 +12,7 @@ from . import conf
 from . import mds
 
 # API compatibility imports
-from .core import NoEventDescriptors, NoRunStop
+from .core import NoEventDescriptors, NoRunStop, NoRunStart
 
 logger = logging.getLogger(__name__)
 

--- a/metadatastore/core.py
+++ b/metadatastore/core.py
@@ -22,6 +22,9 @@ class NoRunStop(Exception):
     pass
 
 
+class NoRunStart(Exception):
+    pass
+
 class NoEventDescriptors(Exception):
     pass
 
@@ -186,7 +189,11 @@ def run_start_given_uid(uid, run_start_col, run_start_cache):
         return run_start_cache[uid]
     except KeyError:
         pass
+
     run_start = run_start_col.find_one({'uid': uid})
+
+    if run_start is None:
+        raise NoRunStart("No runstart with uid {!r}".format(uid))
     return _cache_run_start(run_start, run_start_cache)
 
 

--- a/metadatastore/test/test_commands.py
+++ b/metadatastore/test/test_commands.py
@@ -461,7 +461,7 @@ def test_cache_clear_lookups():
 def test_run_stop_by_run_start():
     run_start_uid, e_desc_uid, data_keys = setup_syn()
     run_stop_uid = mdsc.insert_run_stop(run_start_uid,
-                                       ttime.time(), uid=str(uuid.uuid4()))
+                                        ttime.time(), uid=str(uuid.uuid4()))
     run_start = mdsc.run_start_given_uid(run_start_uid)
     run_stop = mdsc.run_stop_given_uid(run_stop_uid)
     ev_desc = mdsc.descriptor_given_uid(e_desc_uid)
@@ -491,7 +491,7 @@ def test_find_run_start():
 def test_find_run_stop():
     run_start_uid, e_desc_uid, data_keys = setup_syn()
     run_stop_uid = mdsc.insert_run_stop(run_start_uid, ttime.time(),
-                                       uid=str(uuid.uuid4()))
+                                        uid=str(uuid.uuid4()))
 
     run_start = mdsc.run_start_given_uid(run_start_uid)
     run_stop = mdsc.run_stop_given_uid(run_stop_uid)
@@ -513,19 +513,24 @@ def test_find_last_for_smoke():
     last, = mdsc.find_last()
 
 
+@raises(mdsc.NoRunStart)
+def test_fail_runstart():
+    mdsc.run_start_given_uid('aardvark')
+
+
 @raises(ValueError)
 def test_bad_event_desc():
 
-    data_keys = {k:  {'source': k,
-                      'dtype': 'number',
-                      'shape': None} for k in ['foo', 'foo.bar']
+    data_keys = {k: {'source': k,
+                     'dtype': 'number',
+                     'shape': None} for k in ['foo', 'foo.bar']
                  }
     scan_id = 1
 
     # Create a BeginRunEvent that serves as entry point for a run
     rs = mdsc.insert_run_start(scan_id=scan_id, beamline_id='testing',
-                              time=ttime.time(),
-                              uid=str(uuid.uuid4()))
+                               time=ttime.time(),
+                               uid=str(uuid.uuid4()))
 
     # Create an EventDescriptor that indicates the data
     # keys and serves as header for set of Event(s)

